### ISSUE TITLE
Constructor overloads not working with LuaJIT

### ISSUE THREAD

**Issue by porpoises** [ThePhD/sol2#738]
C++ class:

```cpp
struct Foo {
    Foo(int x) { }
    Foo(int x, int y) { }
};
```

Binding:

```cpp
lua.new_usertype<Foo>("Foo", sol::constructors<Foo(int), Foo(int, int)>());
```

Lua code:
```lua
Foo.new(1)
```

Result:
```
sol: no matching function call takes this number of arguments and the specified types
```

Mysteriously, this is LuaJIT only.

**Comment by porpoises** [comment]
FWIW, I'm using LuaJIT 2.1.0-beta3.

**Comment by ThePhD** [comment]
No repro with LuaJIT 2.1.0-beta3.

What compiler, what platform, what defines (No SOL_CHECK_ARGUMENTS?), etc...

**Comment by porpoises** [comment]
Only define is SOL_CXX17_FEATURES.

I'm using MSVC 2017 (19.15.26732.1). I extracted the example from a more contrived example though, let me see if a whittled down version still fails...

**Comment by porpoises** [comment]
Okay, given the definition of `Foo` above:

Fine:
```cpp
    sol::state lua;
    lua.new_usertype<Foo>("Foo", sol::constructors<Foo(int, int)>());
    lua.script("Foo.new(1, 2)");
```

Also fine:
```cpp
    sol::state lua;
    lua.new_usertype<Foo>("Foo", sol::constructors<Foo(int)>());
    lua.script("Foo.new(1)");
```

Not fine:
```cpp
    sol::state lua;
    lua.new_usertype<Foo>("Foo", sol::constructors<Foo(int), Foo(int, int)>());
    lua.script("Foo.new(1)");
```

**Comment by ThePhD** [comment]
Fixed and backported into 2.20.6.

https://github.com/ThePhD/sol2/releases/tag/v2.20.6
